beater/beater.go

```diff
@@ -120,14 +121,27 @@ func (bt *beater) Run(b *beat.Beat) error {
 		// during startup. This might change when APM Server is included in Fleet
 		reloadOnce.Do(func() {
 			defer close(done)
+
+			integrationConfig, err := config.NewIntegrationConfig(ucfg.Config)
+			if err != nil {
+				bt.logger.Error("Could not parse integration configuration from Elastic Agent", err)
+				return
+			}
+
 			// TODO(axw) config received from Fleet should be modified to set data_streams.enabled.
 			var cfg *config.Config
-			cfg, err = config.NewConfig(ucfg.Config, elasticsearchOutputConfig(b))
+			apmServerCommonConfig := integrationConfig.APMServer
+			cfg, err = config.NewConfig(apmServerCommonConfig, elasticsearchOutputConfig(b))
 			if err != nil {
-				bt.logger.Warn("Could not parse configuration from Elastic Agent ", err)
+				bt.logger.Error("Could not parse apm-server configuration from Elastic Agent ", err)
+				return
 			}
+
```

Comment:
If you add the following (untested), you can remove the TODO above. This addresses the issue I mentioned on Slack, where the data streams config will be defined for the initial static config, but not for the dynamic input coming from Agent:

```go
apmServerCommonConfig.Merge(common.MustNewConfigFrom(`{"apm-server.data_streams.enabled": true}`))
```

Reply:
Not sure I follow, when agent starts the server it does it with `-E apm-server.data_streams.enabled=true`, right?